### ISSUE TITLE
[api-extractor] Improve SymbolAnalyzer to support "export * from"

### ISSUE THREAD

**Issue by pgonzal** [Microsoft/web-build-tools#949]
The [SymbolAnalyzer](https://github.com/Microsoft/web-build-tools/blob/434da015a65275da629f4166ce846c54d29a6e1d/apps/api-extractor/src/generators/dtsRollup/SymbolAnalyzer.ts) currently relies on `TypeScriptHelpers.getImmediateAliasedSymbol()` for determining whether a symbol was imported from an external package, and if so what module path was used to import it.

This approach cannot handle `export * from '____'` constructs, because there is no symbol alias that can be used to discover the `export` line.  In API Extractor 6 this problem mostly affects the .d.ts rollups.  For the API report, it simply includes extra definitions that technically belong to an external package.

For API Extractor 7 this problem is more serious, because everything uses the .d.ts rollups engine.

I have a plan to improve the SymbolAnalyzer, but it needs to wait until the initial AE7 branch is merged.

**Comment by pgonzal** [comment]
When the .d.ts rollups encounter this issue, it sometimes produces this error:

```
"Program Bug: The symbol ____ is being imported after it was already registered as non-imported"
```

The reason is that it incorrectly believes the symbol is part of the current project, and then later discovers the inconsistency.

**Comment by pgonzal** [comment]
I encountered this `imported after it was already registered as non-imported` error for a definition like this:

**src/index.ts**
```ts
export { SomeApi } from 'another-package';
```
The error only occurs if `SomeApi` was reachable by some other type alias. Otherwise it simply gets treated as a local reference.  We should include this test case when validating the fix.

**Comment by nvh** [comment]
I encountered this error importing the `EditorState` from `draft-js` and using it in a class
A minimal reproduction looks like this:
```typescript
import { EditorState } from "draft-js"
export class Text {
    editorState: EditorState
}
```
making the editorState private fixes the problem

**Comment by pgonzal** [comment]
> I encountered this error importing the `EditorState` from `draft-js` and using it in a class
> A minimal reproduction looks like this:
> ```typescript
> import { EditorState } from "draft-js"
> export class Text {
>     editorState: EditorState
> }
> ```

This is actually an instance of the other issue https://github.com/Microsoft/web-build-tools/issues/663 which involves the `export { x }` construct.  I'll write my investigation notes over there, since the fixes are different for these two problems.

**Comment by pgonzal** [comment]
Support for the `export * from '____'` construct is implemented in my upcoming PR https://github.com/Microsoft/web-build-tools/pull/1002

**Comment by pgonzal** [comment]
Fixed in PR #1002 except for the special case being tracked by issue https://github.com/Microsoft/web-build-tools/issues/1029
